File: leetcode/#268.js
// question --> https://leetcode.com/problems/missing-number/

/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let sumNaturalNumbers = Math.floor((nums.length * (nums.length + 1)) / 2);
    let sum = 0; 
    for (let index = 0; index < nums.length; index++) {
        sum += nums[index];
    }
    return sumNaturalNumbers - sum;
};

// TC: O(N); SC: O(1)

// end 

File: leetcode/#268.py
# question --> https://leetcode.com/problems/missing-number/

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        sumNaturalNumbers = (len(nums) * (len(nums) + 1)) // 2
        return sumNaturalNumbers - sum(nums) 
        
        # TC: O(N); SC: O(1)
        
# end 